presto-spi/src/main/java/com/facebook/presto/spi/statistics/Range.java

```diff
@@ -0,0 +1,127 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.Decimals;
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.Slice;
+
+import java.util.Objects;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.Decimals.isLongDecimal;
+import static com.facebook.presto.spi.type.Decimals.isShortDecimal;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.RealType.REAL;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static java.lang.Double.isNaN;
+import static java.lang.Double.parseDouble;
+import static java.lang.Float.intBitsToFloat;
+import static java.lang.String.format;
+
+public class Range
+{
+    private final double min;
+    private final double max;
+
+    public Range(double min, double max)
+    {
+        if (isNaN(min)) {
+            throw new IllegalArgumentException("min must not be NaN");
+        }
+        if (isNaN(max)) {
+            throw new IllegalArgumentException("max must not be NaN");
+        }
+        if (max < min) {
+            throw new IllegalArgumentException(format("max must be greater than or equal to min. max: %s. min: %s.", max, min));
+        }
+        this.min = min;
+        this.max = max;
+    }
+
+    public double getMin()
+    {
+        return min;
+    }
+
+    public double getMax()
+    {
+        return max;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Range range = (Range) o;
+        return Double.compare(range.min, min) == 0 &&
+                Double.compare(range.max, max) == 0;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(min, max);
+    }
+
+    @Override
+    public String toString()
+    {
+        return "Range{" +
+                "min=" + min +
+                ", max=" + max +
+                '}';
+    }
+
+    public static Range createFromPrestoTypeValues(Type type, Object min, Object max)
```

Comment:
just create?

Reply:
This is a transitional method that is removed in the very next commit. I just moved it to the `MetastoreHiveStatisticsProvider`, since this is the only class that uses it.